net/host.go

```diff
@@ -170,8 +163,10 @@ func (h *Host) SendSwapMessage(msg Message, id types.Hash) error {
 }
 
 // CloseProtocolStream closes the current swap protocol stream.
-func (h *Host) CloseProtocolStream(id types.Hash) {
-	swap, has := h.swaps[id]
+func (h *Host) CloseProtocolStream(offerID types.Hash) {
+	h.swapMu.Lock()
+	swap, has := h.swaps[offerID]
+	h.swapMu.Unlock()
```

Comment:
minor, but change to `RWMutex`?

Reply:
Yeah, I was thinking that too. It's a RW mutex now.